sql-to-dbsp-compiler/SQL-compiler/src/main/java/org/dbsp/sqlCompiler/compiler/backend/jit/ToJitInnerVisitor.java

```diff
@@ -456,66 +461,75 @@ public VisitDecision preorder(DBSPSomeExpression expression) {
         return VisitDecision.STOP;
     }
 
+    static class FT {
+        /**
+         * The name of the JIT function.
+         */
+        public final String name;
+        /**
+         * Type of result produced by JIT function.
+         */
+        public DBSPTypeBaseType resultType;
+
+        FT(String name, DBSPTypeBaseType resultType) {
+            this.name = name;
+            this.resultType = resultType;
+        }
+    }
+    
+    static Map<String, FT> functionTranslation = new HashMap<String, FT>() {{
+        put("extract_second_Timestamp", new FT("dbsp.timestamp.second", DBSPTypeInteger.SIGNED_64));
+        put("extract_minute_Timestamp", new FT("dbsp.timestamp.minute", DBSPTypeInteger.SIGNED_64));
+        put("extract_hour_Timestamp", new FT("dbsp.timestamp.hour", DBSPTypeInteger.SIGNED_64));
+        put("extract_day_Timestamp", new FT("dbsp.timestamp.day", DBSPTypeInteger.SIGNED_64));
+        put("extract_dow_Timestamp", new FT("dbsp.timestamp.day_of_week", DBSPTypeInteger.SIGNED_64));
+        put("extract_doy_Timestamp", new FT("dbsp.timestamp.day_of_year", DBSPTypeInteger.SIGNED_64));
+        put("extract_isodow_Timestamp", new FT("dbsp.timestamp.iso_day_of_week", DBSPTypeInteger.SIGNED_64));
+        put("extract_week_Timestamp", new FT("dbsp.timestamp.week", DBSPTypeInteger.SIGNED_64));
+        put("extract_month_Timestamp", new FT("dbsp.timestamp.month", DBSPTypeInteger.SIGNED_64));
+        put("extract_year_Timestamp", new FT("dbsp.timestamp.year", DBSPTypeInteger.SIGNED_64));
+        put("extract_isoyear_Timestamp", new FT("dbsp.timestamp.iso_year", DBSPTypeInteger.SIGNED_64));
+        put("extract_quarter_Timestamp", new FT("dbsp.timestamp.quarter", DBSPTypeInteger.SIGNED_64));
+        put("extract_decade_Timestamp", new FT("dbsp.timestamp.decade", DBSPTypeInteger.SIGNED_64));
+        put("extract_century_Timestamp", new FT("dbsp.timestamp.century", DBSPTypeInteger.SIGNED_64));
+        put("extract_millennium_Timestamp", new FT("dbsp.timestamp.millennium", DBSPTypeInteger.SIGNED_64));
+        put("extract_epoch_Timestamp", new FT("dbsp.timestamp.epoch", DBSPTypeInteger.SIGNED_32));
```

Comment:
It's hard to tell why some of these are 32 bits whereas others are 64? `date.day_of_week` is signed 32, and `timestamp.day_of_week` is signed 64 for example.

Reply:
That's how the jit defines them

Reply:
I defined them that way since it's what was in the sql lib files

Reply:
Actually the date functions all return I64 in SQL.
I don't think we ever changed that - this comes from Calcite, so it's not my choice.